sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -590,7 +591,11 @@ object ScalaReflection extends ScalaReflection {
               "cannot be used as field name\n" + walkedTypePath.mkString("\n"))
           }
 
-          val fieldValue = Invoke(inputObject, fieldName, dataTypeFor(fieldType))
+          // primitive take only non-null or struct takes non-null object guarded by isNull
```

Comment:
+1

`Option[Int]`, `Option[String]`, `Option[Tuple1[Int]]`, `Option[Tuple1[String]]` are ok?
If so I'd like you to add the following tests to `ExpressionEncoderSuite`:

```scala
encodeDecodeTest(Option(31), "option of int")
encodeDecodeTest(Option("abc"), "option of string")
encodeDecodeTest(Option.empty[Int], "empty option of int")
encodeDecodeTest(Option.empty[String], "empty option of string")
encodeDecodeTest(Option(Tuple1(31)), "option of tuple1 of int")
encodeDecodeTest(Option.empty[Tuple1[Int]], "empty option of tuple1 of int")
encodeDecodeTest(Option(Tuple1("abc")), "option of tuple1 of string")
encodeDecodeTest(Option.empty[Tuple1[String]], "empty option of tuple1 of string")
```